### DISCUSSION TITLE
Segmentation and playback of audio from custom `AudioStreamTrack`

### DISCUSSION THREAD

**Discussion by soycaporal** [aiortc/aiortc#1006] [Q&A]
Hi there, I'm trying to do some very basic audio manipulation, by:
1. loading from an audio file
2. chunk it to segments
3. Play the chunk on the receive callback

I'm just playing around with the server example:

```
class AudioTransformTrack(AudioStreamTrack):
    kind = "audio"
    _start: float
    _timestamp: int

    def __init__(self, track):
        super().__init__()  # don't forget this!
        self.track = track
        self.samples = int(AUDIO_PTIME * sample_rate)
        self.sound = AudioSegment.from_file(r"./gettysburg10.wav").set_frame_rate(sample_rate)
        self.codec = av.CodecContext.create('pcm_s16le', 'r')
        self.codec.sample_rate = 8000 # sample_rate
        self.codec.channels = 2
        self.audio_samples = 0
        self.chunk_number = 0

    async def recv(self):
        time_start = self.chunk_number * self.samples
        time_end = (self.chunk_number + 1) * self.samples
        sound_chunk = self.sound[time_start:time_end]

        packet = av.Packet(sound_chunk.raw_data)
        frame = self.codec.decode(packet)[0]

        frame.pts = int(self.audio_samples)
        frame.time_base = fractions.Fraction(1, sample_rate)

        self.audio_samples += frame.samples
        self.chunk_number = self.chunk_number+1

        return frame
```
When I add that to my audio track

```
    @pc.on("track")
    def on_track(track):
        if track.kind == "audio":
            local_audio = AudioTransformTrack(
                relay.subscribe(track)
            )
            pc.addTrack(relay.subscribe(local_audio))
            recorder.addTrack(track)
```
I get the audio back, but it sounds very squeaky as if only playing a fraction of the sound. What am I missing here?

I know MediaPlayer is the proper way to play a sound file. I'm trying to create a dynamic audio stream, but adding seconds of silences between audio files. What's the right approach?

Before filing an issue please verify the following:

- [x] Check whether there is already an existing issue for the same topic. 
- [x] Ensure you are actually reporting an issue related to ``aiortc``. The goal
  of the issue tracker is not to provide general guidance about WebRTC or free
  debugging of your code.
- [x] Clearly state whether the issue you are reporting can be reproduced with one
  of the examples provided with ``aiortc`` *without any changes*.
- [x] Be considerate to the maintainers. ``aiortc`` is provided on a best-effort,
  there is no guarantee your issue will be addressed.

**Comment by JesusZerpa** [comment]
Hello, I think you are trying to do something similar to my case, although mine is related to video, try the webcam example, it is half a clue to make my implementation, you may also find the code that I shared in the new case that I have opened https:/ /github.com/aiortc/aiortc/issues/933 there I achieved a dynamic transmission in terms of video, only that I had to modify the library at a specific point because I have not found how to overwrite the functionality because of the private methods that the objects use , I hope it helps you

This is how I make the transmission
```
    pc.addTrack(VideoTransformTrack())
    #pc.addTrack(relay.subscribe(player.video))
    # handle offer
    await pc.setRemoteDescription(offer)
    await recorder.start()


    # send answer
    answer = await pc.createAnswer()
    await pc.setLocalDescription(answer)
```
If you are using webrtc from a browser you may have to modify the javascript so that it is not necessary to send the tracks from your devices, as happens in the server example where the video frame comes from the browser and then you modify it to return it, this is if your data comes only from the server
